utils/bashrc.d/20-ocm-login.bashrc

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+if [[ -z "${OCMC_ENGINE}" ]]
+then
+  # Not launched by ocm-container binary; skip automation
+  # and allow the user to decide when/if this is necessary
+  return
+fi
+
+if ! ocm whoami &> /dev/null
+then
+  ocm login --url="${OCMC_OCM_URL:-https://api.openshift.com}" --use-device-code
+else
+  ocm config set url "${OCMC_OCM_URL}"
+fi
```

Comment:
I worry that this will break the ephemeral status of containers. With this change and the read-write nature of the config mount now we're effectively making it so that in each container instance we share state with every other container instance. So if I log into one container in the prod environment and then spin up another in the staging environment this `ocm config set url` command would overwrite the config that's shared between all containers.

What I think might work is the following:

```suggestion
DEFAULT_OCM_URL="https://api.openshift.com"

if ! ocm whoami &> /dev/null
then
  ocm login --url="${OCMC_OCM_URL:-DEFAULT_OCM_URL}" --use-device-code
fi

export OCM_URL="${OCMC_OCM_URL:-DEFAULT_OCM_URL}"
```

Which would prompt the login if the session isn't current and then export the `OCM_URL` environment variable to override all future OCM API calls.

This would still have the side effect of changing the current session outside of the container, though, for SREs who may use OCM outside of the `ocm-container` environment, though I think this is unlikely to happen due to token expiration times and potentially unlikely to cause any major issues since if they were logged out to begin with....

Example scenario:

* SRE is working in the staging env on their workstation, their session expires at some point after the last usage of `ocm`
* SRE opens `ocm-container` in the prod environment to help with a swarm. Because their session is expired it prompts for a login.
* SRE logs in, finishes their task, and exits the container.
* Going back to whatever they were doing, they now though are logged into the `production` OCM environment.

Reply:
Ah yeah, I did not consider that case.  It's probably worth just handling the login, then, but that's going to take some more code.

I thought the OCM_URL environment variable was no longer used by OCM, at least the way I was reading the error message that popped up on each login.

Let me slap a hold on this and do some more thinking.

Reply:
IIRC the OCM_URL never was used by OCM before, we only ever used it locally. Now, with the 0.1.74 release of [ocm-cli](https://github.com/openshift-online/ocm-cli/releases/tag/v0.1.74) the OCM_URL environment variable _is_ used by the CLI for all commands except `ocm login`, `ocm config set` and `ocm config get`.

"Deprecating" this from `ocm-container` still makes sense, as we want to move towards prefacing all of `ocm-container`'s environment variables with `OCMC` so I have no issues deprecating the `ocm-container` binary's usage of `OCM_URL`. We would, and should, still use that within the container itself though to control the actions of the `ocm` binary inside the container.

I think some of the other changes in this PR are valuable - what would be the lift to keep all other changes except for the read-write config dir changes and the login flow change? Then we can move this PR forward while we think about the login flow?

Reply:
We can merge the PR with just the change from `rw` to `ro` for the mount options.  Nothing else here will break with that change, except that you'll still be required to auth-in every time you start the container, if you haven't already.  I think that functionality will only impact those of us not using OCM outside the container, though.

Reply:
I think this is solved.  OCM login occurs outside of the container, the config is updated with the latest tokens, but the URL is not rewritten in the config (on purpose).  This will prevent non-ocm-container OCM results from changing unexpectedly.

Within ocm-container, the results will also not change, but the OCM_URL env is set per container, so multiple environments can be logged in, ephemerally.  

Changing the env inside the container also works, and is not populated back to other containers or the host OCM.

The ocm.json is mounted RO.